### ISSUE TITLE
Server Permissions wrong

### ISSUE THREAD

**Issue by Ryu945** [qeled/discordie#27]
``client.User.can(Discordie.Permissions.Text.SEND_MESSAGES, e.message.guild)``
returns true whether the bot has permission to send messages or not.

I tested this with channel overrides in every position.

**Comment by qeled** [comment]
Guild context makes it compute permissions from the roles the user has. If you want it to account channel overrides, you should pass channel context, instead of guild.

**Comment by Ryu945** [comment]
I am saying that the guild permissions from roles are wrong.  Don't give a role the send message permission and then run a check on it.  It will say they have the permission.

**Comment by qeled** [comment]
Can you just dump `permissionsFor` output and all roles the account has? Might be just an oversight on your side, there are few cases where permissions are always true, like users with `ADMINISTRATOR` permission have all flags by default. Things like that are stated in the Discord permissions window help bubbles right next to permission flags.

Also you can check if it actually computed a valid permission value by sending a message into a new channel without overwrites.

**Comment by Ryu945** [comment]
I ran this text on my bot permissions using client.User to avoid the server owner issues you mentioned.  Also this role does not have the Administrator or Manage Server permissions.  I also had the bot try to send a message in the channel they calculated they had the permission for and they failed to send it even though they calculated they had the permission for it.  During that text I removed overwrites for that permission so that only server role permissions will determine if they can send a message.  If I understand you correctly you want a ``client.User.permissionsFor(e.message.guild);``
